### DISCUSSION TITLE
Add support for S0-based water/gas meter

### DISCUSSION THREAD

**Discussion by goeran03** [arendst/Tasmota#18421] [Show and tell]
Obviously there no such solution yet. I've added a new sensor working with S0 (pulse) -interface of commen gas/water-meters.
It shows current flow rate and total consumption for max 4 channels. The consumption values are stored to flash.
For setting paratmeters, here is an example:

Config: {"NAME":"S0_WG","GPIO":[0,0,0,2049,0,0,0,0,2048,0,2050,0,0,0],"FLAG":0,"BASE":1}

Parameter setzen:
console: sensor121 **sensor** **cmd** **val**

sensor=1..4

cmd 0= display values (z.B. sensor77 1 0 0)
cmd 1=usage
cmd 2=resolution
cmd 3=type 1=water 2=gas
cmd 4=low debounce [ms]
cmd 5=high debounce [ms]

val=value

{"Gas1":{"Total Usage":5535.800,"Resolution":0.0100,"Low-Debounce":1000,"High-Debounce":1000}}
{"Water2":{"Total Usage":10.920,"Resolution":0.0010,"Low-Debounce":50,"High-Debounce":50}}
{"Water3":{"Total Usage":1.251,"Resolution":0.0010,"Low-Debounce":50,"High-Debounce":50}}
Use following command:
Backlog sensor121 1 3 2;sensor121 1 2 0.01 ;sensor121 1 1 6875.220;sensor121 1 4 1000;sensor121 1 5 1000
Backlog sensor121 2 3 1;sensor121 2 2 0.001;sensor121 2 1 66.562  ;sensor121 2 4 50;  sensor121 2 5 50
Backlog sensor121 3 3 1;sensor121 3 2 0.001;sensor121 3 1 2.227   ;sensor121 3 4 50;  sensor121 3 5 50

**Comment by sfromis** [comment]
Actually, the Smart Meter Interface does AFAIK support S0 pulses.

**Reply by goeran03** [reply]
only just 1 channel and not storing total consumption values over power ccycle, if I'm right.
